src/main/scala/ayai/components/QuestBag.scala

```diff
@@ -14,10 +14,35 @@ case class QuestBag(quests: ArrayBuffer[Quest] = new ArrayBuffer[Quest]()) exten
   }
 
   def addQuest(questToAdd: Quest): Unit = {
-    println("Quest with ID: " + questToAdd.id + " Added to Bag" )
-
     if (questToAdd != null) {
+      println("Quest with ID: " + questToAdd.id + " Added to Bag" )
+      println( "   " + questToAdd.asJson() )
+
       quests += questToAdd
     }
   }
+
+  def removeQuest(id: Int) : Quest = {
+    println("Quest with ID: " + id + " Removed from Bag" )
+
+    for(quest <- quests) {
+      if (quest.id == id) {
+        return quests.remove( quests.indexOf(quest) )
+      }
+    }
+    return null
```

Comment:
`Option[T]` should be favored over `null` in scala (prevents all types of hairiness)